### PR TITLE
refactor(themes): implement Indigo Theme changes

### PR DESCRIPTION
**Pull request by simeonoff**
- [x] Avatar
- [x] Badge
- [x] Button
- [x] Button Group
- [x] Card
- [x] Checkbox
- [x] Dialog
- [x] Expansion Panel
- [x] Icon
- [x] Icon Button
- [x] Input
- [x] Radio
- [x] Snackbar
- [x] Textarea

### CODE REVIEW

**Comment by andiesm813** [comment]
Avatar
I don't know how to set the type icon, just make sure that in Light Mode the color of the icon is Grays.600.
Everything else is fine.

**Comment by andiesm813** [comment]
**Checkbox**
Since i wasnt able to check the angular ones, maybe the same issues are also happening (some were already mentioned by Silvana.

**Both**

- Label font size should be Body 2.
- Make sure the check BOX is 16x16px. I see it a bit smaller, about 15px high and 16px wide. It's causing the outer stroke to look uneven.

image

- Invalid state outer focused stroke seems to be thicker than the design and it should be set to Error400 50% color.

image

**Light Theme**

- Invalid label color in default state should be Grays.800
- Invalid label color in Hover state should be Grays.900

**Dark Theme**

- Invalid label color in default state should be White 80%.
- Invalid label color in Hover state should be White 100%.
- Error color default (of the box background) should be Error.400

**Comment by andiesm813** [comment]
**Dialog**

- The font of the message should be Body 1
- The size of the buttons should be Medium
- In Dark Mode the elevation should be 23


Also both elevation styles that light and dark dialogs are using need to be updated to the following values:

**Shadow 24 (light theme)**
These should be the default settings of the shadow style:
box-shadow:0 0px 16px 0 #24252c66 (grays.900 40%)

**Shadow 23 (dark theme)**
These should be the default settings of the shadow style:
box-shadow:0 0px 16px 0 #0000004d (Black 30%)

**Comment by andiesm813** [comment]
**Button**
The icon color of the contained button in idle state should be white 80%. It looks like it is in the code... but if i snoop the implemented component, i see it white.

image

**Comment by andiesm813** [comment]
**Card**

- I know users can change the style of the buttons and whatnot, but by design, the avatar type is set to circle. Can we default to that? 
- And of the action button (the text one), if it has only one, let's keep it contained and if it has 2, one contained (the more important one) and one outlined in some of the samples. So that users may know what the default. Then they may change to whatever they want.
- For the icon action buttons, let's keep them FLAT by default.

**Comment by andiesm813** [comment]
**Snackbar**
No visual discrepancies

**Comment by andiesm813** [comment]
**Radio Group**

- Radio Group Horizontal: space between separate radio buttons should be 16px
- The color of the Disabled Checked (filled) variants in Light and Dark Mode should be Primary 400 with 50% opacity
- The DOT of the radio item is slightly horizontally misaligned. It's a BIT shifted to the left. This happens in bootstrap too. I dont know why. But it's happening.

image

**Comment by andiesm813** [comment]
Input Group
I cannot see the search type. There was a requested fix in the Angular PR that should be applied here as well:
For "searchbox" (input type "search"), the gap (2) should be 8px and it also has 8px left and right inner padding (1).

image

**Comment by SisIvanova** [comment]
> **Input Group**
> 
> * I cannot see the search type. There was a requested fix in the Angular PR that should be applied here as well:
> 
> For "searchbox" (input type "search"), the gap (2) should be 8px and it also has 8px left and right inner padding (1).
> 
> image
> 
> * The --filled-text-hover-color property does not exist in the light theme (in the styles) and it's not really working in the dark theme (it's supposed to be changing to White, but it's not really changing).
> 
> image
> 
> * Light and Dark themes: --focused-text-color should be the same as --filled-text-hover-color (grays900 /white 100%).

You better delete this "undefined" string and then type your text, I believe this is a sample issue.

**Comment by SisIvanova** [comment]
> **Button** The icon color of the contained button in idle state should be white 80%. It looks like it is in the code... but if i snoop the implemented component, i see it white.
> 
> image

The "+" and "-" are not icons that's why the color is not correct. You can check the last button sample to see the icon color.

**Comment by andiesm813** [comment]
**Checkbox**
I still see 2 issues with the checkbox.
The "box" is not really square.... If you see this red square around it, that would be a perfect square... and if you look at the checkbox inside, there is more white space on the left of the box.
And in the focused state, that issue is more noticeable, because there's like half a pixel of white space between the BOX and the outer focused border.

image


The other issue is the vertical alignment of the label. It seems to be vertically misaligned.... like a bit pushed to the top....

image

**Comment by andiesm813** [comment]
> **Card**
> 
> * I know users can change the style of the buttons and whatnot, but by design, the avatar type is set to circle. Can we default to that?
> * And of the action button (the text one), if it has only one, let's keep it contained and if it has 2, one contained (the more important one) and one outlined in some of the samples. So that users may know what the default. Then they may change to whatever they want.
> * For the icon action buttons, let's keep them FLAT by default.
> * The styling of the elevations needs to be updated according to what's in the figma file. The elevations used in card are: --ig-elevation-4 and --ig-elevation-6
> * When all content is hidden, except for actions, the top padding should be also 16px. Currently it's 8px.
> 
> image
> 
> * Is the font size of the content different on purpose or is it in the actual component in this card type? if it's for sample purposes it's ok.... if the default font size of this card type is that, it needs to be like the rest of the card types:
> 
> image

I still think that some of the examples should show the default style too. But i can survive.
However, there seems to be a difference between the ELEVATED card in ANGULAR and this WC one.
The angular also has a border color, which is the expected style. Cards in AppBuilder always have stroke... and the elevated ones keep the stroke and add the box-shadow.
So, whatever was done in Angular has to be replicated here as well. And it would be nice if the component is consistent in both platforms. All buttons in Angular are FLAT while in WC are CONTAINED.

image

**Comment by andiesm813** [comment]
> **Radio Group**
> 
> * Radio Group Horizontal: space between separate radio buttons should be 16px
> * The color of the Disabled Checked (filled) variants in Light and Dark Mode should be Primary 400 with 50% opacity
> * The DOT of the radio item is slightly horizontally misaligned. It's a BIT shifted to the left. This happens in bootstrap too. I dont know why. But it's happening.
> 
> image

The 3rd issue is still happening. 

image

**Comment by andiesm813** [comment]
> **Input Group**
> 
> * I cannot see the search type. There was a requested fix in the Angular PR that should be applied here as well:
> 
> For "searchbox" (input type "search"), the gap (2) should be 8px and it also has 8px left and right inner padding (1).
> 
> image
> 
> * The --filled-text-hover-color property does not exist in the light theme (in the styles) and it's not really working in the dark theme (it's supposed to be changing to White, but it's not really changing).
> 
> image
> 
> * Light and Dark themes: --focused-text-color should be the same as --filled-text-hover-color (grays900 /white 100%).

The last 2 issues are still visible. There are those 2 properties missing in the css, while they do exist in Angular:

image

**Comment by andiesm813** [comment]
**Icon Button**
The disabled border color is not working. It is available in the css though. 

image

**Comment by andiesm813** [comment]
**Textarea**
The same issues as the input group are also happening here. There are colors set in the Angular input group (which affect the text area) that are not present in the WC text area:

image

**Comment by SisIvanova** [comment]
> > **Input Group**
> > 
> > * I cannot see the search type. There was a requested fix in the Angular PR that should be applied here as well:
> > 
> > For "searchbox" (input type "search"), the gap (2) should be 8px and it also has 8px left and right inner padding (1).
> > image
> > 
> > * The --filled-text-hover-color property does not exist in the light theme (in the styles) and it's not really working in the dark theme (it's supposed to be changing to White, but it's not really changing).
> > 
> > image
> > 
> > * Light and Dark themes: --focused-text-color should be the same as --filled-text-hover-color (grays900 /white 100%).
> 
> The last 2 issues are still visible. There are those 2 properties missing in the css, while they do exist in Angular:
> 
> image

Screenshot 2024-03-13 at 10 02 04 AM
Screenshot 2024-03-13 at 10 02 44 AM
Screenshot 2024-03-13 at 10 03 51 AM
Screenshot 2024-03-13 at 10 04 13 AM

**Comment by SisIvanova** [comment]
> > **Radio Group**
> > 
> > * Radio Group Horizontal: space between separate radio buttons should be 16px
> > * The color of the Disabled Checked (filled) variants in Light and Dark Mode should be Primary 400 with 50% opacity
> > * The DOT of the radio item is slightly horizontally misaligned. It's a BIT shifted to the left. This happens in bootstrap too. I dont know why. But it's happening.
> > 
> > image
> 
> The 3rd issue is still happening.
> 
> image

Not sure how are you testing the radio and the checkbox but we can't reproduce this issue. Make sure you haven't zoomed in your browser or you haven't changed your browser's default resolution. It is kind of weird that you are testing with a 20px box having in mind that the radio is 16px.

MicrosoftTeams-image (6)

**Comment by andiesm813** [comment]
> **Card:**
> 
> * I think something is broken here, because I saw in previous screenshots the gap is okay. It should be 8px.
> 
> image
> * In the following examples there's 16px padding under card content and then there are additional 16px padding on top of the actions. I believe this isn't correct and the bottom padding of the content should be 8px.
> 
> image

There is no gap property set, that i can see in the CSS. If i set it, it looks ok, no matter the amount of text in the content. 

image

**Comment by sbayreva** [comment]
**Checkbox**
1. Light and Dark Mode- Disabled indeterminate color - the color is the right one ( Primary 400 with 50% opacity) but it does not appear in the exaple in storybook
Screenshot 2024-03-15 at 10 41 26
2.I cannot see style for the tick and the dash (indeterminate state) when checkbox is Disabled. The color should be white with 40% opacity for Light and Dark Mode.

**Comment by imincheva** [comment]
**Snackbar:**
- There's truncation after the second line, which is incorrect and not present in Angular.
image

**Comment by AnjiManova** [comment]
**Icon**

No discrepancies, and it's already checked as done.

**Comment by sbayreva** [comment]
**Radio Button**
1. The font should be linked to Body 2 style

**Comment by imincheva** [comment]
**Input:**
- Label should be linked with caption text style. It's not linked in Angular neither, so you might not want to change it (same thing I mentioned for the Badge)
- The Label color seems to not be linked, as the color variable is defined correctly. I can't inspect to find where exactly the problem comes from, but it should be the same color and look exactly as the Label. Applies for both light and dark mode.

**Comment by AnjiManova** [comment]
**Button**

Contained - all good as colors and spacings

Outlined
- Light & Dark Themes: Disabled state -> The border should be 2px (0.125rem)

- When we have an Outlined button and a Contained button, if the content is the same, both button widths should be the same, but now the border in the Outlined buttons adds 4 additional pixels to the button width.

Flat - all good as colors and spacings

FAB
Large size should have 10px left and right paddings
Medium size should have 8px left and right paddings
Small size should have 6px left and right padding

**Comment by AnjiManova** [comment]
**Icon Button**

Contained:
- Dark Theme & Light Theme -> On hover the Icon should be white 100%

**Comment by SisIvanova** [comment]
> **Card:**
> 
> * In the following examples there's 16px padding under card content and then there are additional 16px padding on top of the actions. I believe this isn't correct and the bottom padding of the content should be 8px.
> 
> image

This is a limitation that we cannot handle because in this sample the card header, content, and actions are wrapped in an additional `div` element in order to achieve this design.

**Comment by AnjiManova** [comment]
### **Button Group**

**Light Theme**
Unchecked button:
- Active state should be without the focus border
- Focused & Active - border should be grays.500

Checked button:
- Focused - item border color shouldn't be grays.400 -> maybe primary.500?
- Active state is not correctly linked/added or something. When we click on the button we are activating the Focused mode which I think is not the expected behaviour.
- Focused & Active is styled properly but I'm not able to force the state from the inspect mode. If this is expected behaviour ignore this comment

**Dark Theme**
Unchecked button:
- Active state should be without the focus border
- Focused & Active - border should be grays.200

Checked button:
- Focused - item border color shouldn't be grays.100 -> maybe primary.500?
- Active state is not correctly linked/added or something. When we click on the button we are activating the Focused mode which I think is not the expected behaviour.
- Focused & Active is styled properly but I'm not able to force the state from the inspect mode. If this is expected behaviour ignore this comment

**Comment by SisIvanova** [comment]
> **Radio Button**
> 
> 1. The font should be linked to Body 2 style

The radio button label typography is correct. It is applied to `[part='label']`.

**Comment by SisIvanova** [comment]
> **Checkbox**
> 
> 1. Light and Dark Mode- Disabled indeterminate color - the color is the right one ( Primary 400 with 50% opacity) but it does not appear in the exaple in storybook
> 
> Screenshot 2024-03-15 at 10 41 26
> 
> 2.I cannot see a style for the tick and the dash (indeterminate state) when checkbox is Disabled. The color should be white with 40% opacity for Light and Dark Mode. 
> 3. Required and Invalid states in Storybook cannot be triggered by the true-false toggles 
> 4. The font should be linked to Body 2 style

The checkbox label typography is correct. It is applied to `[part='label']`. 
The rest of the comments should be fixed.

**Comment by AnjiManova** [comment]
Everything with the Button and Button Group looks fine as implemented styles. Two comments left, but they are not related only to Indigo, so if they are going to be implemented, they should be added as separate issues/features and I am going to check the Button and Button Group as done.

In addition, the comments are:
- Button Group: Active state should be without the focus border (This behaviour is applied only in Fluent and should be logged as a separate feature/issue for Indigo, Material & Bootstrap)
- Button: When we have an Outlined button and a Contained button, if the content is the same, both button widths should be the same, but now the border in the Outlined buttons adds four additional pixels to the button width. (this behaviour is in all themes, not only Indigo)

**Comment by SisIvanova** [comment]
> Everything with the Button and Button Group looks fine as implemented styles. Two comments left, but they are not related only to Indigo, so if they are going to be implemented, they should be added as separate issues/features and I am going to check the Button and Button Group as done.
> 
> In addition, the comments are:
> 
> * Button Group: Active state should be without the focus border (This behaviour is applied only in Fluent and Indigo and should be logged as a separate feature/issue for Material & Bootstrap). This is applicable for the Button component as well.
> * Button: When we have an Outlined button and a Contained button, if the content is the same, both button widths should be the same, but now the border in the Outlined buttons adds four additional pixels to the button width. (this behaviour is in all themes, not only Indigo)

As we discussed earlier these changes will be made in separate PR-s.

File: src/components/avatar/themes/shared/avatar.indigo.scss [hunk 1]
```diff
@@ -5,4 +5,10 @@ $theme: $indigo;
 
 :host {
     --component-size: var(--ig-size, #{var-get($theme, 'default-size')});
+
+    ::slotted(igc-icon) {
+        --ig-size: 1;
```
Comment (simeonoff) [inline src/components/avatar/themes/shared/avatar.indigo.scss:10]: When possible, use `--component-size` instead of `--ig-size`;

**Comment by AnjiManova** [comment]
**Textarea**

The label font style should be linked to the Caption style if it is, ignore this comment and just let us know to check the Textarea as done.

**Review by SisIvanova — APPROVED** [review]

**Pull request merged**